### DISCUSSION TITLE
is it possible to use operators with the entity query?

### DISCUSSION THREAD

**Discussion by sombriks** [vincentlauvlwj/Ktorm#385] [Q&A] [answered]
i was looking for some idiom like this:

val employees = database.sequenceOf(Employees).filter { (it.name like "%$search%")  or (it.jobTitle like "%$search%") }.toList()

I saw https://www.ktorm.org/en/entity-finding.html#Get-Entities-by-Query-DSL, just looking for something different

**Comment by vincentlauvlwj** [comment]
Yes, of course.

**Reply by sombriks** [reply]
so how this code snipped can become valid? https://github.com/sombriks/rosetta-beer-store/blob/336add6e5a72298787b3b58d8cc5da656bcd5001/beer-store-service-kotlin-javalin-ktorm-liquibase/src/main/kotlin/beer/store/controllers/BeerController.kt#L25

**Comment by sombriks** [comment]
https://github.com/sombriks/rosetta-beer-store/pull/1/files rn it says it's invalid, how can i proceed?

**Comment by vincentlauvlwj** [comment, marked as answer]
It's a problem of Kotlin compiler, note the position of line breaking.

Fail: 

```kotlin
val rows = Db.database.sequenceOf(Beers).filter {
    (it.descriptionBeer like "%$search%")
    or (it.titleBeer like "%$search%")
}
```

Works: 
```kotlin
val rows = Db.database.sequenceOf(Beers).filter {
    (it.descriptionBeer like "%$search%") or
    (it.titleBeer like "%$search%")
}
```

Also works: 
```kotlin
val rows = Db.database.sequenceOf(Beers).filter {
    (it.descriptionBeer like "%$search%")
    .or(it.titleBeer like "%$search%")
}
```

**Reply by sombriks** [reply]
i was so close, thank you so much!
